apps/src/templates/Dialog.jsx

```diff
@@ -138,8 +152,24 @@ const Dialog = React.createClass({
     onCancel: whenNoChildOfTypes(Buttons),
     confirmText: whenNoChildOfTypes(Buttons),
     onConfirm: whenNoChildOfTypes(Buttons),
+    confirmType: whenNoChildOfTypes(Buttons),
   }),
 
+  handleKeyDown(event) {
+    // Always focus the Cancel or Confirm button when tab is pressed, to prevent the
+    // user from selecting elements outside of the dialog.
+    if (event.key === 'Tab') {
+      const cancelButton = this.refs.buttons.getCancelButton();
+      const confirmButton = this.refs.buttons.getConfirmButton();
+      if ((document.activeElement === cancelButton)) {
+        confirmButton && confirmButton.focus();
+      } else {
+        cancelButton && cancelButton.focus();
+      }
+      event.preventDefault();
```

Comment:
It will not work in your example. How about this behavior:
* add "cancelButton" as a class within the Cancel component (rather than from here within the Dialog class)
* tab through all buttons that are in the dialog, without relying on there being a "cancelButton"
* if no button is selected and a "cancelButton" exists, start with the first one of those you find. Otherwise start with the first button you find.